### DISCUSSION TITLE
ESP32-CAM support

### DISCUSSION THREAD

**Discussion by Geoide** [Moddable-OpenSource/moddable#922] [General]
Hello.

Is there some support to the ESP32-CAM. I would like to watch the video from this device to a ILI9341 TFT LCD Screen Display but I can't find the way to.

Thanks.

**Comment by andycarle** [comment]
Hi @Geoide!

To the best of my knowledge, no one has worked with the Moddable SDK on the ESP32-CAM development kit specifically. But we have worked with a lot of the same pieces.

The OV2640 camera hardware module on the ESP32-CAM is the same module as is used in our [Arducam module](https://github.com/Moddable-OpenSource/moddable/tree/public/modules/drivers/arducam) and examples (e.g. [arducampreview](https://github.com/Moddable-OpenSource/moddable/tree/public/examples/drivers/arducampreview). I have also done some work with the same camera module included in the [Kaluga](https://docs.espressif.com/projects/esp-idf/en/latest/esp32s2/hw-reference/esp32s2/user-guide-esp32-s2-kaluga-1-kit.html) kit. So, it should definitely be possible to get working.

The main variation among places where the OV2640 shows up is on how the camera module is wired. And that's where I'm a little fuzzy on the ESP32-CAM board... Do you happen to have a preferred reference on how the camera module is connected to the ESP32 on the ESP32-CAM board? I've seen a fair number of references to working with the ESP32-CAM as an external module running the built-in firmware. But less about how to interact with the camera itself using ESP-IDF code running on the ESP32-CAM's ESP32. If we could pull together all of that information, I can make a more concrete recommendation about how to proceed.

&nbsp;- Andy
